scalafix-cli/src/main/scala/scalafix/cli/CliRunner.scala

```diff
@@ -91,24 +97,63 @@ sealed abstract case class CliRunner(
     display.stop()
     exitCode.get()
   }
+
+  // checks if outFile contents have changed since the creation its .semanticdb file.
+  private def isUpToDate(input: Input, outFile: File): Boolean =
+    if (database.isEmpty) true
+    else if (!outFile.exists() && cli.outTo.nonEmpty) true
+    else {
+      input match {
+        case Input.LabeledString(_, _) =>
+          val fileToWrite = scala.io.Source.fromFile(outFile)
+          val originalContents = input.chars.toIterator
+          try {
+            @tailrec
+            def isEqual(a: Iterator[Char], b: Iterator[Char]): Boolean =
+              if (a.hasNext && b.hasNext) a.next() == b.next() && isEqual(a, b)
+              else if (a.hasNext || b.hasNext) false
+              else true
+            isEqual(fileToWrite, originalContents)
+          } finally fileToWrite.close()
```

Comment:
@xeno-by It's unnecessary to load everything into memory when we can do it char by char.